src/core/trtserver.cc

```diff
@@ -626,6 +626,41 @@ TrtServerResponse::OutputData(
   return nullptr;  // Success
 }
 
+//
+// TrtServerModelIndex
+//
+// Implementation for TRTSERVER2_ModelIndex.
+//
+class TrtServerModelIndex {
+ public:
+  TrtServerModelIndex() = default;
+  TRTSERVER_Error* GetModels(
+      const char* const** models, uint64_t* models_count);
+  ni::ModelRepositoryIndex model_repository_index_;
+
+ private:
+  std::vector<const char*> indices_;
+};
+
+TRTSERVER_Error*
+TrtServerModelIndex::GetModels(
```

Comment:
GetModelNames? This can just return pointers directly into model_repository_index_ strings. Don't need to create indices_.